src/spokes/BalanceSheet.sol

```diff
@@ -160,9 +181,19 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
         emit Issue(poolId, scId, to, _pricePoolPerShare(poolId, scId), shares);
 
         if (queueEnabled[poolId][scId]) {
-            queuedShares[poolId][scId].increase += shares;
+            if (queuedShares[poolId][scId].isPositive || queuedShares[poolId][scId].delta == 0) {
+                queuedShares[poolId][scId].delta += shares;
+                queuedShares[poolId][scId].isPositive = true;
+            } else if (queuedShares[poolId][scId].delta > shares) {
+                queuedShares[poolId][scId].delta -= shares;
+                queuedShares[poolId][scId].isPositive = false;
+            } else {
+                queuedShares[poolId][scId].delta = shares - queuedShares[poolId][scId].delta;
+                queuedShares[poolId][scId].isPositive = true;
+            }
```

Comment:
I think the logic is perfect!

Maybe we can factorize the maths into a `DeltaMathLib.sol` where we define a `Delta` struct with `add/sub` associated method? That way, we reduce the complexity at this contract level for the readers. Easy to test from the lib too.

Reply:
Good idea!

Reply:
This is actually a bit challenging. We have 2 other properties on `ShareQueueAmount` struct, which are not generic. So we cannot just use a `Delta` struct.

We could add methods to convert to/from `ShareQueueAmount` and `Delta`, and then do the computations on `Delta`. But that seems like more overhead.